nginx: build falls with missing dependencie libpcre.so.1

Maintainer: @heil @Ansuel 
Environment: 
Target arch - x86_64
OpenWRT version - master, lib musl
Build os: ubuntu server 18.04
GCC: 10.x
Description:
It looks like the other packages that use libpcre are builded correctly
Steps to reproduce:
1. ./scrips/feeda update -a && ./scrips/feeda install -a -f
2. In menuconfig select to * nginx-all-module, nginx-ssl-util, nginx-mod-luci
3. make V=s
4. Profit

This is a part of build log:
```
install -m0600 ./files/_redirect2ssl.conf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/.pkgdir/nginx-all-module/etc/nginx/conf.d/
touch /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/.pkgdir/nginx-all-module.installed
mkdir -p /media/openwrt/bin/targets/x86/64/packages /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/CONTROL /media/openwrt/staging_dir/target-x86_64_musl/pkginfo
install -d -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/usr/sbin
install -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/usr/sbin/nginx /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/usr/sbin/
install -d -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/conf.d
install -m0644 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/mime.types /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/koi-utf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/koi-win /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/win-utf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/fastcgi_params /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-install/etc/nginx/uwsgi_params /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/
install -m0600 ./files/nginx.conf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/
install -m0600 ./files/_lan.conf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/conf.d/
install -d -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/init.d
install -m0755 ./files/nginx.init /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/init.d/nginx
install -d -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx
install -m0755 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/nginx-naxsi/naxsi_config/naxsi_core.rules /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx
chmod 0640 /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/naxsi_core.rules
install -m0600 ./files/_redirect2ssl.conf /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx/etc/nginx/conf.d/
find /media/openwrt/build_dir/target-x86_64_musl/nginx-all-module/nginx-1.19.2/ipkg-x86_64/nginx -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
Package nginx is missing dependencies for the following libraries:
libpcre.so.1
Makefile:561: recipe for target '/media/openwrt/bin/packages/x86_64/packages/nginx_1.19.2-1_x86_64.ipk' failed
make[3]: *** [/media/openwrt/bin/packages/x86_64/packages/nginx_1.19.2-1_x86_64.ipk] Error 1
make[3]: Leaving directory '/media/openwrt/feeds/packages/net/nginx'
time: package/feeds/packages/nginx/all-module/compile#87.25#11.22#95.05
package/Makefile:111: recipe for target 'package/feeds/packages/nginx/compile' failed
make[2]: *** [package/feeds/packages/nginx/compile] Error 2
make[2]: Leaving directory '/media/openwrt'
package/Makefile:107: recipe for target '/media/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile' failed
make[1]: *** [/media/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/media/openwrt'
/media/openwrt/include/toplevel.mk:233: recipe for target 'world' failed
make: *** [world] Error 2

```

--- a/net/nginx/Makefile
+++ b/net/nginx/Makefile
@@ -81,10 +81,7 @@ define Package/nginx/default
   SUBMENU:=Web Servers/Proxies
   TITLE:=Nginx web server
   URL:=http://nginx.org/
-  DEPENDS:=+NGINX_PCRE:libpcre +libopenssl \
-	+NGINX_PCRE:nginx-ssl-util +!NGINX_PCRE:nginx-ssl-util-nopcre \
-	+NGINX_HTTP_GZIP:zlib +NGINX_LUA:liblua +libpthread +NGINX_DAV:libxml2 \
-	+NGINX_UBUS:libubus +NGINX_UBUS:libblobmsg-json +NGINX_UBUS:libjson-c
+  DEPENDS:=+libopenssl +libpthread
   PROVIDES:=nginx
 endef
 
@@ -97,6 +94,10 @@ define Package/nginx-ssl
   $(Package/nginx/default)
   TITLE += with SSL support
   VARIANT:=ssl
+  DEPENDS+= +NGINX_PCRE:libpcre \
+	+NGINX_PCRE:nginx-ssl-util +!NGINX_PCRE:nginx-ssl-util-nopcre \
+	+NGINX_HTTP_GZIP:zlib +NGINX_LUA:liblua +libpthread +NGINX_DAV:libxml2 \
+	+NGINX_UBUS:libubus +NGINX_UBUS:libblobmsg-json +NGINX_UBUS:libjson-c
   CONFLICTS:=nginx-all-module
 endef
 
@@ -107,8 +108,8 @@ Package/nginx-ssl/description = $(Package/nginx/description) \
 define Package/nginx-all-module
   $(Package/nginx/default)
   TITLE += with ALL module selected
-  DEPENDS:=+libpcre +libopenssl +zlib +liblua +libpthread +libxml2 \
-	+libubus +libblobmsg-json +libjson-c +nginx-ssl-util
+  DEPENDS+=+libpcre +nginx-ssl-util +zlib +liblua +libxml2 +libubus \
+	+libblobmsg-json +libjson-c
   VARIANT:=all-module
   PROVIDES += nginx-ssl
 endef
@@ -286,12 +287,15 @@ else
   CONFIG_NGINX_LUA:=y
   CONFIG_NGINX_DAV:=y
   CONFIG_NGINX_UBUS:=y
-  ADDITIONAL_MODULES += --add-module=$(PKG_BUILD_DIR)/nginx-naxsi/naxsi_src \
-    --add-module=$(PKG_BUILD_DIR)/lua-nginx --with-ipv6 --with-http_stub_status_module --with-http_flv_module \
-	--with-http_dav_module --add-module=$(PKG_BUILD_DIR)/nginx-dav-ext-module \
+  ADDITIONAL_MODULES += --with-ipv6 --with-http_stub_status_module --with-http_flv_module \
+	--with-http_dav_module \
 	--with-http_auth_request_module --with-http_v2_module --with-http_realip_module \
-	--with-http_secure_link_module --with-http_sub_module --add-module=$(PKG_BUILD_DIR)/nginx-headers-more \
+	--with-http_secure_link_module --with-http_sub_module \
 	--with-stream --with-stream_ssl_module --with-stream_ssl_preread_module \
+	--add-module=$(PKG_BUILD_DIR)/nginx-headers-more \
+	--add-module=$(PKG_BUILD_DIR)/nginx-naxsi/naxsi_src \
+	--add-module=$(PKG_BUILD_DIR)/lua-nginx \
+	--add-module=$(PKG_BUILD_DIR)/nginx-dav-ext-module \
 	--add-module=$(PKG_BUILD_DIR)/nginx-brotli --add-module=$(PKG_BUILD_DIR)/nginx-rtmp \
 	--add-module=$(PKG_BUILD_DIR)/nginx-ts --add-module=$(PKG_BUILD_DIR)/nginx-ubus-module
   config_files += koi-utf koi-win win-utf fastcgi_params uwsgi_params
@@ -541,23 +545,18 @@ ifeq ($(CONFIG_NGINX_UBUS),y)
   endef
 endif
 
-$(eval $(call BuildPackage,nginx-ssl))
-$(eval $(call BuildPackage,nginx-all-module))
-$(eval $(call BuildPackage,nginx-mod-luci))
-
-
 # TODO: remove after a transition period (together with pkg nginx-util):
 # It is for smoothly substituting nginx and nginx-mod-luci-ssl (by nginx-ssl
 # respectively nginx-mod-luci). Add above commented PROVIDES when removing.
 
 Package/nginx = $(Package/nginx-ssl)
-
 Package/nginx/install = $(Package/nginx-ssl/install)
-
 Package/nginx/prerm = $(Package/nginx-ssl/prerm)
-
 $(eval $(call BuildPackage,nginx))
 
+$(eval $(call BuildPackage,nginx-ssl))
+$(eval $(call BuildPackage,nginx-all-module))
+$(eval $(call BuildPackage,nginx-mod-luci))
 
 define Package/nginx-mod-luci-ssl
   TITLE:=Dummy package for transition when upgrading.

